gpytorch/math/functions/lazy_kronecker_product/kp_toeplitz_mvn_kl_divergence.py

```diff
@@ -0,0 +1,126 @@
+
+        # log |K2|
+        log_det_covar2, = StochasticLQ(num_random_probes=10).evaluate(mul_closure,
+                                                                      len(mu_diff),
+                                                                      [lambda x: x.log()])
+
+        # Tr(K2^{-1}K1)
+        sample_matrix = torch.sign(torch.randn(self.num_samples, len(mu_diff)))
+        trace = 0
+        for z in sample_matrix:
+            trace = trace + quad_form_closure(z)
+        trace = trace / self.num_samples
+
+        # Inverse quad form
+        mat_inv_y = LinearCG().solve(mul_closure, mu_diff)
+        inv_quad_form = mat_inv_y.dot(mu_diff)
+
+        res = log_det_covar2 + trace + inv_quad_form
+
+        self.save_for_backward(mu_diff, chol_covar1, covar2_toeplitz_columns)
+        self.mul_closure = mul_closure
+        self.mat_inv_y = mat_inv_y
+
+        return mu_diff.new().resize_(1).fill_(res)
+
+    def backward(self, grad_output):
+        grad_output_value = grad_output.squeeze()[0]
+
+        mu_diff, chol_covar1, covar2_toeplitz_columns = self.saved_tensors
+        mat_inv_y = self.mat_inv_y
+        mul_closure = self.mul_closure
+
+        def deriv_quad_form_closure(z):
+            I_minus_Kinv_M_z = z - LinearCG().solve(mul_closure, chol_covar1.t().mv(chol_covar1.mv(z)))
+            Kinv_z = LinearCG().solve(mul_closure, z)
+            return kp_sym_toeplitz_derivative_quadratic_form(covar2_toeplitz_columns, Kinv_z, I_minus_Kinv_M_z)
+
+        grad_mu_diff = None
+        grad_cholesky_factor = None
+        grad_toeplitz_columns = None
+
+        if self.needs_input_grad[0]:
+            # Need gradient with respect to mu_diff
+            grad_mu_diff = mat_inv_y.mul_(2 * grad_output_value)
+
+        if self.needs_input_grad[1]:
+            # Compute gradient with respect to the Cholesky factor L
+            grad_cholesky_factor = 2 * LinearCG().solve(mul_closure, chol_covar1)
+            grad_cholesky_factor.mul_(grad_output_value)
+
+        if self.needs_input_grad[2]:
+            sample_matrix = torch.sign(torch.randn(self.num_samples, len(mu_diff)))
+
+            grad_toeplitz_columns = torch.zeros(covar2_toeplitz_columns.size())
+            for z in sample_matrix:
+                grad_toeplitz_columns = grad_toeplitz_columns + deriv_quad_form_closure(z)
+            grad_toeplitz_columns = grad_toeplitz_columns / self.num_samples
+
+            grad_toeplitz_columns.add_(- kp_sym_toeplitz_derivative_quadratic_form(grad_toeplitz_columns,
+                                                                                   mat_inv_y.squeeze(),
+                                                                                   mat_inv_y.squeeze()))
+            grad_toeplitz_columns.mul_(grad_output_value)
+
+        return grad_mu_diff, grad_cholesky_factor, grad_toeplitz_columns
+
+
+class KPToeplitzMVNKLDivergence(Function):
+    """
```

Comment:
We got rid of the `MVNKLDivergence` Functions in the latest changes to master. See what's in `ToeplitzLazyVariable` now.